module/x/gravity/abci.go

```diff
@@ -100,12 +102,30 @@ func pruneSignerSetTxs(ctx sdk.Context, k keeper.Keeper) {
 		earliestToPrune := currentBlock - params.SignedSignerSetTxsWindow
 		for _, set := range k.GetSignerSetTxs(ctx) {
 			if set.Nonce < lastObserved.Nonce && set.Height < earliestToPrune {
+				k.DeleteEthereumSignatures(ctx, set.GetStoreIndex())
 				k.DeleteOutgoingTx(ctx, set.GetStoreIndex())
 			}
 		}
 	}
 }
 
+// pruneEventVoteRecords deletes all event vote records with nonces that are older than the last observed event nonce
+func pruneEventVoteRecords(ctx sdk.Context, k keeper.Keeper) {
+	lastObservedEventNonce := k.GetLastObservedEventNonce(ctx)
```

Comment:
This is only semi-related, but we should add a query endpoint for last observed event nonce, currently it's only available on per-validator basis.

Reply:
made https://github.com/PeggyJV/gravity-bridge/issues/523